client/dashboard/domains/domain-contact-verification/index.tsx

```diff
@@ -0,0 +1,235 @@
+import { domainWhoisQuery, domainContactVerificationMutation } from '@automattic/api-queries';
+import { useSuspenseQuery, useMutation } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Card,
+	CardBody,
+	FormFileUpload,
+	Button,
+	Notice,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { domainContactInfoRoute, domainRoute } from '../../app/router/domains';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { SectionHeader } from '../../components/section-header';
+import { findRegistrantWhois } from '../../utils/domain';
+
+const MAX_FILE_SIZE = 5 * 1024 * 1024; // 5MB
+
+export default function DomainContactVerification() {
+	const { domainName } = domainRoute.useParams();
+	const [ selectedFiles, setSelectedFiles ] = useState< FileList | null >( null );
+	const { data: domainWhois } = useSuspenseQuery( domainWhoisQuery( domainName ) );
+	const registrantWhoisData = findRegistrantWhois( domainWhois );
+	const [ submitted, setSubmitted ] = useState( false );
+	const [ error, setError ] = useState( false );
+	const { mutate: domainContactVerification, isPending: isSubmitting } = useMutation(
+		domainContactVerificationMutation( domainName )
+	);
+	const { createErrorNotice, createSuccessNotice } = useDispatch( noticesStore );
+	const router = useRouter();
+
+	if ( ! registrantWhoisData ) {
+		return null;
+	}
+
+	const handleFilesSelected = ( event: React.ChangeEvent< HTMLInputElement > ) => {
```

Comment:
I think we should wrap this function within useCallback to prevent unnecessary re-renders of child components that receive these functions as props.

Reply:
I looked into this, and in this specific case I'm not sure `useCallback` is necessary since `handleFilesSelected` is passed directly to a native DOM element via the `FormFileUpload` component 🤔. I think the overhead would outweigh any gains here.